### PR TITLE
Update README to reflect deprecation of cuSignal

### PR DESCRIPTION
**Pull request by awthomp**
Deprecation Announcement for cuSignal: https://docs.rapids.ai/notices/rsn0032/

cuSignal source is moving to CuPy, and RAPIDS 23.08 is the last formal release with cuSignal. 

Merge after https://github.com/rapidsai/cusignal/pull/587

### CODE REVIEW

**Comment by jakirkham** [comment]
Thanks Adam! 🙏

File: README.md [hunk 1]
```diff
@@ -1,5 +1,15 @@
 # <div align="left"><img src="https://rapids.ai/assets/images/rapids_logo.png" width="90px"/>&nbsp;cuSignal</div>
 
+# Deprecation Notice
+RAPIDS v23.08 is the last formal release for cuSignal. Starting in 2019, cuSignal strived to provide a simple to use and highly performant library for signal processing developers working in Python. With over 400,000 total Anaconda downloads, we built an incredible community together. We are pleased to announce that much of the cuSignal codebase is moving to [CuPy](https://cupy.dev) and is highly aligned with CuPy's goal to GPU-accelerate the NumPy and SciPy ecosystems. By moving our work to CuPy, we allow developers to use a single library for signal processing (and more!) while also reducing software dependencies.
+
+To continue your journey with GPUs, Python, and Signal Processing, please:
+- [Download CuPy](https://docs.cupy.dev/en/stable/install.html)
+- Visit `cupyx.scipy.signal` [source code](https://github.com/cupy/cupy/tree/main/cupyx/scipy/signal)
```
Comment (leofang) [inline README.md:8]:
```suggestion
- Visit `cupyx.scipy.signal` [source code](https://github.com/cupy/cupy/tree/main/cupyx/scipy/signal) and [documentation](https://docs.cupy.dev/en/latest/reference/scipy_signal.html)
```

**Review by BradReesWork — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (awthomp) [reply README.md:8]: Great suggestion @leofang. I'll include this as a new bullet point in the "what's next" list.

**Review by mnicely — APPROVED** [review]

**Pull request merged**